hamocc/mo_vgrid.F90

```diff
@@ -3,334 +3,303 @@
-                                          ! this are ignored and set to the concentration of the 
-                                          ! layer above). Note that the bottom layer index kbo(i,j)
-                                          ! is defined as the lowermost layer thicker than dp_min_sink.
-
-  INTEGER, DIMENSION(:,:),   ALLOCATABLE :: kmle
-  INTEGER, DIMENSION(:,:),   ALLOCATABLE :: kbo
-  INTEGER, DIMENSION(:,:),   ALLOCATABLE :: kwrbioz
-  INTEGER, DIMENSION(:,:),   ALLOCATABLE :: k0100,k0500,k1000,k2000,k4000
-  REAL,    DIMENSION(:,:),   ALLOCATABLE :: bolay
-  REAL,    DIMENSION(:,:,:), ALLOCATABLE :: ptiestu
-  REAL,    DIMENSION(:,:,:), ALLOCATABLE :: ptiestw
-
-contains
-!******************************************************************************
 
+  !******************************************************************************
+  ! Variables and routines related to vertical grid  structure
+  ! Declaration, memory allocation, and routines related to the
+  ! vertical grid structure. These have to be recalculated every
+  ! time step when iHAMOCC is coupled to BLOM.
+  !
+  !  J.Schwinger,        *NORCE Climate, Bergen*    2020-05-19
+  !******************************************************************************
 
+  implicit none
+  private
+
+  ! Routines
+
+  public :: set_vgrid       ! Calculate variables related to the vertical grid structure.
+  public :: alloc_mem_vgrid ! Allocate memory for vertical grid variables
+
+  ! Module variables
+
+  integer, parameter, public :: kmle_static = 2  ! k-end index for layers that represent the mixed layer in blom.
+
+  ! Default value used for isopycnic coordinates.
+  real,    parameter, public :: dp_ez  = 100.0    ! depth of euphotic zone
+  real,    parameter, public :: dp_min = 1.0e-12  ! min layer thickness layers thinner
+                                                  ! than this are ignored by HAMOCC
+  real,    parameter, public :: dp_min_sink = 1.0 ! min layer thickness for sinking (layers thinner than
+                                                  ! this are ignored and set to the concentration of the
+                                                  ! layer above). note that the bottom layer index kbo(i,j)
+                                                  ! is defined as the lowermost layer thicker than dp_min_sink.
+
+  integer, dimension(:,:),   allocatable, public :: kmle
+  integer, dimension(:,:),   allocatable, public :: kbo     ! number of wet cells in column.
+  integer, dimension(:,:),   allocatable, public :: kwrbioz ! last k-index of euphotic zone.
+  real,    dimension(:,:),   allocatable, public :: bolay   ! height of bottom cell.
+  real,    dimension(:,:,:), allocatable, public :: ptiestu ! depth of layer centres.
+  real,    dimension(:,:,:), allocatable, public :: ptiestw ! depth of layer interfaces.
+  integer, dimension(:,:),   allocatable, public :: k0100
```

Comment:
Could you add the former comment: `k-index of gridbox comprising xxxx m depth`

Reply:
I don't see this comment in the previous code. I have added the following:
  ```
integer, dimension(:,:),   allocatable, public :: k0100   ! k-index of gridbox comprising 100 m depth
  integer, dimension(:,:),   allocatable, public :: k0500   ! k-index of gridbox comprising 500 m depth
  integer, dimension(:,:),   allocatable, public :: k1000   ! k-index of gridbox comprising 1000 m depth
  integer, dimension(:,:),   allocatable, public :: k2000   ! k-index of gridbox comprising 2000 m depth
  integer, dimension(:,:),   allocatable, public :: k4000   ! k-index of gridbox comprising 4000 m depth

```

Reply:
Perfectly fine as well :-)